src/main/java/core/basesyntax/Fruits.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class Fruits {
+    private Map<String, Integer> fruitQuantities;
+
+    public Fruits() {
+        fruitQuantities = new HashMap<>();
+    }
+
+    public int balance(String fruitType,int init) {
+        fruitQuantities.put(fruitType,init);
+        return init;
+    }
+
+    public int supply(String fruitType,int quantityOfFruits) {
+        int supplyAmount = fruitQuantities.get(fruitType) + quantityOfFruits;

```

Comment:
Potential `NullPointerException` if `fruitType` does not exist in `fruitQuantities`. Consider using `getOrDefault(fruitType, 0)` to handle uninitialized fruit types.